src/util.c

```diff
@@ -2447,6 +2447,49 @@ bool nats_IsSubjectValid(const char *subject, bool wcAllowed)
     return true;
 }
 
+// Bitmap of allowed ASCII characters (0–127)
+// Allowed: A–Z, a–z, 0–9, '-', '_', '/', '='
+static const uint8_t _limited_term_allowed_bitmap[16] = {
+    0x00, // 0x00–0x07
+    0x00, // 0x08–0x0F
+    0x00, // 0x10–0x17
+    0x00, // 0x18–0x1F
+    0x00, // 0x20–0x27
+    0x08, // 0x28–0x2F: '/' (0x2F)
+    0xFF, // 0x30–0x37: '0'–'7'
+    0x03, // 0x38–0x3F: '8', '9', '=' (0x3D)
+    0xFE, // 0x40–0x47: 'A'–'G'
+    0xFF, // 0x48–0x4F: 'H'–'O'
+    0xFF, // 0x50–0x57: 'P'–'W'
+    0x03, // 0x58–0x5F: 'X', 'Y', 'Z', '_' (0x5F), '-' (0x2D)
+    0x00, // 0x60–0x67: '`' (not allowed), 'a'–'g'
+    0xFF, // 0x68–0x6F: 'h'–'o'
+    0xFF, // 0x70–0x77: 'p'–'w'
+    0x03  // 0x78–0x7F: 'x', 'y', 'z'
+};
+
+// Validates limited_term, see
+// https://github.com/nats-io/nats-architecture-and-design/blob/8693d2263e137e23748e3df583fd58247e4e48b7/adr/ADR-6.md?plain=1#L48
+natsStatus
+nats_validateLimitedTerm(const char *name, const char *term)
+{
+    int i = 0;
+    int len = (int) strlen(term);
```

Comment:
As it is in this PR, one could call this function with `term` being `NULL`, which would cause this to abort. So I would replace the line below with:
```
if (nats_IsStringEmpty(term))
    return ...

len = (int) strlen(term);
```
here we know that `len` will be > 0.